python/pylibcugraph/pylibcugraph/graphs.pyx

```diff
@@ -0,0 +1,169 @@
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_free,
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+    cugraph_sg_graph_create,
+    cugraph_graph_properties_t,
+    cugraph_sg_graph_free,
+)
+from pylibcugraph.resource_handle cimport (
+    EXPERIMENTAL__ResourceHandle,
+)
+from pylibcugraph.graph_properties cimport (
+    EXPERIMENTAL__GraphProperties,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    assert_CAI_type,
+)
+
+
+cdef class EXPERIMENTAL__SGGraph(EXPERIMENTAL__Graph):
+    """
+    RAII-stye Graph class for use with single-GPU APIs that manages the
+    individual create/free calls and the corresponding cugraph_graph_t pointer.
+
+    Parameters
+    ----------
+    resource_handle : ResourceHandle
+        Handle to the underlying device resources needed for referencing data
+        and running algorithms.
+
+    graph_properties : GraphProperties
+        Object defining intended properties for the graph.
+
+    src_array : device array type
+        Device array containing the vertex identifiers of the source of each
+        directed edge. The order of the array corresponds to the ordering of the
+        dst_array, where the ith item in src_array and the ith item in dst_array
+        define the ith edge of the graph.
+
+    dst_array : device array type
+        Device array containing the vertex identifiers of the destination of each
+        directed edge. The order of the array corresponds to the ordering of the
+        src_array, where the ith item in src_array and the ith item in dst_array
+        define the ith edge of the graph.
+
+    weight_array : device array type
+        Device array containing the weight values of each directed edge. The
+        order of the array corresponds to the ordering of the src_array and
+        dst_array arrays, where the ith item in weight_array is the weight value
+        of the ith edge of the graph.
+
+    store_transposed : bool
+        Set to True if the graph should be transposed. This is required for some
+        algorithms, such as pagerank.
+
+    renumber : bool
+        Set to True to indicate the vertices used in src_array and dst_array are
+        not appropriate for use as internal array indices, and should be mapped
+        to continuous integrers starting from 0.
+
+    do_expensive_check : bool
+        If True, performs more extensive tests on the inputs to ensure
+        validitity, at the expense of increased run time.
+    """
+    def __cinit__(self,
+                  EXPERIMENTAL__ResourceHandle resource_handle,
+                  EXPERIMENTAL__GraphProperties graph_properties,
+                  src_array,
+                  dst_array,
+                  weight_array,
+                  store_transposed,
+                  renumber,
+                  expensive_check):
```

Comment:
Naming mismatch. `do_expensive_check` in the above documentation. `expensive_check` here.

Reply:
fixed in next commit.